cloudinit/distros/networking.py

```diff
@@ -0,0 +1,137 @@
+import abc
+
+from cloudinit import net
+
+
+# Type aliases (https://docs.python.org/3/library/typing.html#type-aliases),
+# used to make the signatures of methods a little clearer
+DeviceName = str
+NetworkConfig = dict
+
+
+class Networking(metaclass=abc.ABCMeta):
+    """The root of tne Networking hierarchy in cloud-init.
+
+    This is part of an ongoing refactor in the cloud-init codebase, for more
+    details see "``cloudinit.net`` -> ``cloudinit.distros.networking``
+    Hierarchy" in HACKING.rst for full details.
+    """
+
+    def _get_current_rename_info(self) -> dict:
+        return net._get_current_rename_info()
+
+    def _rename_interfaces(self, renames: list, *, current_info=None) -> None:
+        return net._rename_interfaces(renames, current_info=current_info)
+
+    def apply_network_config_names(self, netcfg: NetworkConfig) -> None:
+        return net.apply_network_config_names(netcfg)
+
+    def device_devid(self, devname: DeviceName):
+        return net.device_devid(devname)
+
+    def device_driver(self, devname: DeviceName):
```

Comment:
Can we raise an `NotImplementedError` by default instead. We cannot support that on NetBSD, on possible OpenBSD too.

Reply:
You can't infer the device driver in use given a network interface name? (I was under the impression from another/previous comment that the driver name was encoded in the name itself, e.g. rtk0 is using the rtk driver?)

Reply:
Yes, this is the way, but it's rather empirical, and AFAIR on FreeBSD the interface name can be changed, and in this case, you cannot get the original name.

Reply:
you can, but it's harder